### PR TITLE
Fix PSES crash on debug start when function breakpoint defined

### PR DESCRIPTION
**Pull request by rkeithhill**
Fixex https://github.com/PowerShell/vscode-powershell/issues/1159

When VSCode passes us a breakpoint to set, we normally set
a flag to indicate "setBreakpointInProgress" so that when
the DebugService_BreakpointUpdated event is fired, we can
tell that we initiated it instead of the user using Set-PSBreakpoint
to set a breakpoint.   Well, the code that handled function
breakpoints msgs sent by VSCode was not setting that flag.

Also, when the user does use Set-PSBreakpoint -Command
there is no debug protocol event for function breakpoints
so we need to ignore this type of breakpoint set by the user
until the debug protocol support it.  
See https://github.com/Microsoft/vscode-debugadapter-node/issues/157

### CODE REVIEW

**Comment by rkeithhill** [comment]
Hmm, all the Microsoft.PowerShell.EditorServices.Test.dll tests pass on my machine:
```
=== TEST EXECUTION SUMMARY ===
   Microsoft.PowerShell.EditorServices.Test  Total: 93, Errors: 0, Failed: 0, Skipped: 3, Time: 12.497s
~\GitHub\rkeithhill\PowerShellEditorServices\test\PowerShellEditorServices.Test
```

**Comment by tylerl0706** [comment]
AppVeyor likes to throw a fit every once in a while. We're good now :)

File: src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs [hunk 1]
```diff
@@ -632,6 +646,24 @@ protected void Stop()
             RequestContext<object> requestContext)
         {
             // TODO: Handle this appropriately
+            //if (!this.noDebug)
+            //{
+            //    this.setBreakpointInProgress = true;
+
+            //    try
+            //    {
+            //        // Set exception breakpoints in DebugService
```
Comment (tylerl0706) [inline src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs]: Any reason we need this commented out code? If so, can you supply a reason in the comment?

Comment (rkeithhill) [reply src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs]: I'm hoping we will eventually get support for breaking on unhandled exceptions (maybe even first thrown).  When we do, I don't want to have the same mistake of not setting the setBreakpointInProgress flag.  That was the one of the causes of the crash.  I can add a comment along those lines.

**Comment by rkeithhill** [comment]
@tylerl0706 Can you kick the appveyor build?

**Comment by tylerl0706** [comment]
kicked ⚽

**Pull request merged**